ArduCopter/control_auto.cpp

```diff
@@ -324,6 +324,13 @@ void Copter::auto_land_run()
 
     // process pilot's input
     if (!failsafe.radio) {
+        if((g.throttle_behavior & THR_BEHAVE_HIGH_THROTTLE_CANCELS_LAND) != 0 && rc_throttle_control_in_filter.get() > 0.7f*THR_MAX){
+            // exit land if throttle is high
+            if (!set_mode(LOITER)) {
```

Comment:
For clarity, this should read:
bool success = set_mode(LOITER);
if (!success) { set_mode(ALT_HOLD); }